src/iterable-functions.php

```diff
@@ -28,6 +29,21 @@ function iterable_map(iterable $iterable, callable $mapper): iterable
     return iterable($iterable)->map($mapper);
 }
 
+/**
+ * Merge iterables
+ *
+ * @param iterable<TKey, TValue> ...$args
```

Comment:
BTW I think this template will not work for merging iterables with different value types.

Also it seems, there's no support for typing it properly https://github.com/phpstan/phpstan/issues/3814#issuecomment-684779628